### PR TITLE
Cleanup up Workflow logs

### PR DESCRIPTION
**Pull request by deepakwadhwani**
- Cleanup workflow logs
- Cleanup some logs for Spark
- Local Job Runner to log rootcause exception as error.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java [hunk 1]
```diff
@@ -60,22 +61,22 @@ public void configure() {
 
   @Override
   public void run() {
+    String prettyProgramType = programType != null ? ProgramType.valueOf(programType.name()).getPrettyName() : "";
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java]: I dont think `programType` can be null

Comment (deepakwadhwani) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java]: Sure. I am just using the existing approach, this just being the change for logs. Will check further.

File: cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java [hunk 2]
```diff
@@ -559,7 +560,8 @@ public void run() {
           LOG.warn("Job {} was killed", id, t);
         } else {
           this.status.setRunState(JobStatus.FAILED);
-          LOG.error("Job {} failed", id, t);
+          LOG.error("Job Id '{}' failed. Error: ", id.getId(), ExceptionUtils.getRootCause(t));
+          LOG.debug("Job Id '{}' failed. Complete Error: ", id.getId(), t);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java]: Why log two stack traces?

Comment (deepakwadhwani) [reply cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java]: @chtyim the idea was to only log the root cause stack trace as error, but still have the complete available in the debugs. Please let me know your thoughts.

Comment (deepakwadhwani) [reply cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java]: Hi @chtyim I have removed the other one, please take another look.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java (hunk 1 above)
Comment (deepakwadhwani) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java]: @anwar6953 Doesnt seem like programType can be NULL. Not sure why the existing code was checking for that. DUT ran ok with the change: https://builds.cask.co/browse/CDAP-DUT5698-3

File: cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java [hunk 3]
```diff
@@ -559,7 +560,7 @@ public void run() {
           LOG.warn("Job {} was killed", id, t);
         } else {
           this.status.setRunState(JobStatus.FAILED);
-          LOG.error("Job {} failed", id, t);
+          LOG.error("Job Id '{}' failed. Error: ", id.getId(), ExceptionUtils.getRootCause(t));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/org/apache/hadoop/mapred/LocalJobRunnerWithFix.java]: Use `Throwables.getRootCause` instead.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java [hunk 4]
```diff
@@ -60,22 +61,22 @@ public void configure() {
...
       programRunner.run();
 
       // TODO (terence) : Put something back to context.
 
-      LOG.info("{} Program {} workflow action completed",
-               programType != null ? programType.name() : null, programName);
+      LOG.info("{} Program '{}' in workflow completed", prettyProgramType, programName);
     } catch (Exception e) {
-      LOG.info("Failed to execute {} Program {} in workflow", programType, programName, e);
+      LOG.error("Failed to execute {} Program '{}' in workflow", prettyProgramType, programName, e);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java]: in general, shouldn't log an exception if it is being throw. I think we can just get rid of this `try-catch` block.

**Comment by deepakwadhwani** [comment]
@chtyim addressed your comments, please take another look.

**Comment by chtyim** [comment]
LGTM. Please resolve the conflicts. Note that spark classes are moved to `cdap-spark-core-base`.

**Comment by deepakwadhwani** [comment]
@chtyim I did not see any conflicts, the diffs were after those changes I believe.
Build running: https://builds.cask.co/browse/CDAP-DUT5698-6

**Pull request merged**